lib/matplotlib/tests/test_path.py

```diff
@@ -49,6 +49,12 @@ def test_contains_points_negative_radius():
     np.testing.assert_equal(result, [True, False, False])
 
 
+def test_exact_extents_cubic():
+    hard_curve = Path([[0, 0], [1, 0], [1, 1], [0, 1]],
+                      [Path.MOVETO, Path.CURVE4, Path.CURVE4, Path.CURVE4])
+    assert(hard_curve.get_extents().bounds == (0., 0., 0.75, 1.))
+
+
```

Comment:
Do we want more tests of other bezier curves in here?  

I guess this work is part of a larger work, but probably still needs a "Whats New" entry.  

I think the PR would be helped by explaining what we returned before this PR.  What does `hard_curve.get_extents` return now if anything?  Just the control points?

Reply:
It used to approximate the bezier curve as a polyline (with "high" resampling).

Reply:
OK, so is this actually a performance improvement then?  And do we get the same answer plus or minus the interpolation?  I have no way of knowing if this PR is "correct".

Reply:
Yes, it just returns the bbox of the control points.

Reply:
And sorry for the double reply, I'm happy to add a more comprehensive list of Bezier curves, I think it would be nice to have a more comprehensive test suite, that hits several corner cases (cubic curves that are representable as quadratic curves, degenerate curves, etc).

Reply:
So, does this PR cause a major performance hit for "normal" plots (as opposed to someone who has hand built a Bezier path for some reason)?  I'm a little ignorant if this all gets called for every marker in a `scatter` plot.

Reply:
Good question. That was the initial reason why the original PR actually defined a new function "get_exact_extents" and just changed the docstring of the existing "get_extents" to tell the truth (that it doesn't work for curves exactly). 

However, I've dug around more now, and it seems like, there's explicit code that prevents calling this a few places, for example, at all as long as clip is set on, markers don't have this method called at all.

Should I post timing tests? The tl;dr is there's really not a difference. For a very underwhelming (but honestly representative of standard usage) example, 11 of the tests in `test_path.py` use this function, but there is *zero* measurable slowdown on running those tests.

Reply:
> It used to approximate the bezier curve as a polyline (with "high" resampling).

I gather from reading code comments and previous issues that it did this at some point, but it does not do that anymore, or else there's a bug (see #16830).

Reply:
Possibly this was lost at some point, may be worth (or not...) doing a bit of repo archeology.

Reply:
If it was lost, it was lost before #7184, so has been gone for quite a long time. Probably not worth digging for it IMO...

Reply:
I think it is adequate to just show for a few bezier curves that this implementation is "correct".

If this is indeed not called if clip is on, then I imagine it is not a performance concern.